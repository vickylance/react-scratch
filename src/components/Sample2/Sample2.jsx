import React from 'react';
import PropTypes from 'prop-types';
import { Title } from './Sample2.style';

const Sample = ({ firstName = '', lastName = '' }) => {
  return <Title>Hi {`${firstName} ${lastName}`}!</Title>;
};

Sample.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
};

Sample.defaultProps = {
  firstName: '',
  lastName: '',
};

export default Sample;
