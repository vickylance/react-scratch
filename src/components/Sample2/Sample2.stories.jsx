import React from 'react';

import Sample2 from './Sample2';

export const Welcome = () => <Sample2 firstName="Vignesh" lastName="S" />;

export const WithoutLastname = () => <Sample2 firstName="Vignesh" />;

export default {
  title: 'Example/Sample2',
  component: Sample2,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
};
