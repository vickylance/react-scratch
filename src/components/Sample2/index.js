import Sample2 from './Sample2';

export default Sample2;
