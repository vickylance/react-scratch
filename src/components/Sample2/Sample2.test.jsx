import React from 'react';
import { render, screen } from '@testing-library/react';
import Sample2 from './Sample2';

describe('This will test MyComponent', () => {
  it('renders correctly', () => {
    const { container } = render(<Sample2 firstName="Alejandro" lastName="Roman" />);
    expect(container).toMatchSnapshot();
  });

  it('renders message', () => {
    render(<Sample2 firstName="Alejandro" lastName="Roman" />);
    expect(screen.getByText('Hi Alejandro Roman!')).toBeInTheDocument();
  });
});
