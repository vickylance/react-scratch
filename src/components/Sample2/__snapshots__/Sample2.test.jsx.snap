// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This will test MyComponent renders correctly 1`] = `
<div>
  <h1
    class="sc-bdvvaa kNZsls"
  >
    Hi 
    Alejandro Roman
    !
  </h1>
</div>
`;
